src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -35,6 +35,20 @@ end
 ##### `Diagonal`
 #####
 
+_diagview(x::Diagonal) = x.diag
+_diagview(x::AbstractMatrix) = view(x, diagind(x))
+_diagview(x::Tangent{<:Diagonal}) = x.diag
+function ChainRulesCore.rrule(::typeof(sqrt), d::Diagonal)
+    y = sqrt(d)
+    @assert y isa Diagonal
+    function sqrt_pullback(Δ_raw)
+      Δ = unthunk(Δ_raw)
+      Δ_diag = _diagview(Δ)
+      return NoTangent(), Diagonal(Δ_diag ./ (2 .* y.diag))
```

Comment:
I would collapse these just in the name of readability, and I think it wants 4 spaces:
```suggestion
    function sqrt_pullback(Δ)
        Δ_diag = _diagview(unthunk(Δ))
        return NoTangent(), Diagonal(Δ_diag ./ (2 .* y.diag))
```